typing/typeopt.ml

```diff
@@ -243,8 +243,16 @@ let array_kind_of_elt ~elt_sort env loc ty =
     else
       Pgcscannableproductarray (scannable_product_array_kind loc sorts)
   in
+  (* CR dkalinichenko: layout, externality and separability should be
```

Comment:
This isn't enough to detect `float array`, I think. But I like the general idea here.

Reply:
Oh, right.